From ab5655ef3e9727ad26e7c095d79a64ce4e8b7b28 Mon Sep 17 00:00:00 2001
From: Mauro Carvalho Chehab <mchehab+samsung@kernel.org>
Date: Thu, 14 Mar 2019 16:52:49 -0300
Subject: [PATCH] mediawidget: fix a regression with next/previous logic

A fix at the CD handing code broke for other sources.

After this fix, both prev/next buttons work and also the DBUS
equivalent logic:

	qdbus org.mpris.kaffeine /Player Prev
	qdbus org.kde.kaffeine /Player Prev

Thanks to Antoni Bella for identifying the broken commit and
addressing it.

BUG: 405276

Signed-off-by: Mauro Carvalho Chehab <mchehab+samsung@kernel.org>
---
 src/mediawidget.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/mediawidget.cpp b/src/mediawidget.cpp
index 91fd961..c34e9c5 100644
--- a/src/mediawidget.cpp
+++ b/src/mediawidget.cpp
@@ -794,14 +794,14 @@ void MediaWidget::previous()
 {
 	if (source->getType() == MediaSource::Url)
 		emit playlistPrevious();
-	backend->jumpToPreviousChapter();
+       source->previous();
 }
 
 void MediaWidget::next()
 {
 	if (source->getType() == MediaSource::Url)
 		emit playlistNext();
-	backend->jumpToNextChapter();
+       source->next();
 }
 
 void MediaWidget::stop()
-- 
2.21.0

